### PR TITLE
[DOCS] Note clause limit in `index.mapping.total_fields.limit` docs

### LINKED ISSUE #46928: [DOCS] _exists_ returns `too_many_clauses` exception
**Elasticsearch version**: 7.3.2
**Plugins installed**: none
**JVM version**: 12.0.2
**OS version**: latest docker container for 7.3.2

**Issue**
Running the following query: `{"query": {"exists": {"field": "myField"}}}}` returns a "all shards failed" due to `too_many_clauses` exception.

The container log reports:
```
elasticsearch_1  | {"type": "server", "timestamp": "2019-09-20T13:49:09,025+0000", "level": "DEBUG", "component": "o.e.a.s.TransportSearchAction", "cluster.name": "elasticsearch", "node.name": "es01", "cluster.uuid": "Lj6To9uwSTetSdntuC4Wyw", "node.id": "iCm50_K2QEq8aV9FZcrTWg",  "message": "All shards failed for phase: [query]" , 
elasticsearch_1  | "stacktrace": ["org.elasticsearch.index.query.QueryShardException: failed to create query: {",
elasticsearch_1  | "  \"exists\" : {",
elasticsearch_1  | "    \"field\" : \"myField\",",
elasticsearch_1  | "    \"boost\" : 1.0",
elasticsearch_1  | "  }",
elasticsearch_1  | "}",
elasticsearch_1  | "at org.elasticsearch.index.query.QueryShardContext.toQuery(QueryShardContext.java:321) ~[elasticsearch-7.3.2.jar:7.3.2]",
elasticsearch_1  | "at org.elasticsearch.index.query.QueryShardContext.toQuery(QueryShardContext.java:304) ~[elasticsearch-7.3.2.jar:7.3.2]",
elasticsearch_1  | "at org.elasticsearch.search.SearchService.parseSource(SearchService.java:743) ~[elasticsearch-7.3.2.jar:7.3.2]",
elasticsearch_1  | "at org.elasticsearch.search.SearchService.createContext(SearchService.java:591) ~[elasticsearch-7.3.2.jar:7.3.2]",
elasticsearch_1  | "at org.elasticsearch.search.SearchService.createAndPutContext(SearchService.java:550) ~[elasticsearch-7.3.2.jar:7.3.2]",
elasticsearch_1  | "at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:353) ~[elasticsearch-7.3.2.jar:7.3.2]",
elasticsearch_1  | "at org.elasticsearch.search.SearchService.lambda$executeQueryPhase$1(SearchService.java:340) ~[elasticsearch-7.3.2.jar:7.3.2]",
elasticsearch_1  | "at org.elasticsearch.action.ActionListener.lambda$map$2(ActionListener.java:145) ~[elasticsearch-7.3.2.jar:7.3.2]",
elasticsearch_1  | "at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62) [elasticsearch-7.3.2.jar:7.3.2]",
elasticsearch_1  | "at org.elasticsearch.search.SearchService$2.doRun(SearchService.java:1052) [elasticsearch-7.3.2.jar:7.3.2]",
elasticsearch_1  | "at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [elasticsearch-7.3.2.jar:7.3.2]",
elasticsearch_1  | "at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44) [elasticsearch-7.3.2.jar:7.3.2]",
elasticsearch_1  | "at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:758) [elasticsearch-7.3.2.jar:7.3.2]",
elasticsearch_1  | "at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [elasticsearch-7.3.2.jar:7.3.2]",
elasticsearch_1  | "at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]",
elasticsearch_1  | "at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]",
elasticsearch_1  | "at java.lang.Thread.run(Thread.java:835) [?:?]",
elasticsearch_1  | "Caused by: org.apache.lucene.search.BooleanQuery$TooManyClauses: maxClauseCount is set to 1024",
elasticsearch_1  | "at org.apache.lucene.search.BooleanQuery$Builder.add(BooleanQuery.java:114) ~[lucene-core-8.1.0.jar:8.1.0 dbe5ed0b2f17677ca6c904ebae919363f2d36a0a - ishan - 2019-05-09 19:34:03]",
elasticsearch_1  | "at org.apache.lucene.search.BooleanQuery$Builder.add(BooleanQuery.java:127) ~[lucene-core-8.1.0.jar:8.1.0 dbe5ed0b2f17677ca6c904ebae919363f2d36a0a - ishan - 2019-05-09 19:34:03]",
elasticsearch_1  | "at org.elasticsearch.index.query.ExistsQueryBuilder.newObjectFieldExistsQuery(ExistsQueryBuilder.java:208) ~[elasticsearch-7.3.2.jar:7.3.2]",
elasticsearch_1  | "at org.elasticsearch.index.query.ExistsQueryBuilder.newFieldExistsQuery(ExistsQueryBuilder.java:195) ~[elasticsearch-7.3.2.jar:7.3.2]",
elasticsearch_1  | "at org.elasticsearch.index.query.ExistsQueryBuilder.newFilter(ExistsQueryBuilder.java:157) ~[elasticsearch-7.3.2.jar:7.3.2]",
elasticsearch_1  | "at org.elasticsearch.index.query.ExistsQueryBuilder.doToQuery(ExistsQueryBuilder.java:130) ~[elasticsearch-7.3.2.jar:7.3.2]",
elasticsearch_1  | "at org.elasticsearch.index.query.AbstractQueryBuilder.toQuery(AbstractQueryBuilder.java:99) ~[elasticsearch-7.3.2.jar:7.3.2]",
elasticsearch_1  | "at org.elasticsearch.index.query.QueryShardContext.lambda$toQuery$1(QueryShardContext.java:305) ~[elasticsearch-7.3.2.jar:7.3.2]",
elasticsearch_1  | "at org.elasticsearch.index.query.QueryShardContext.toQuery(QueryShardContext.java:317) ~[elasticsearch-7.3.2.jar:7.3.2]",
elasticsearch_1  | "... 16 more"] }
```

**Steps to reproduce**:
The issue relies on the mapping structure of the `myField` property, which contains ~80 sub-properties, and those have between 10 to 50 (or more) sub-properties.
No analysis is defined for the index.

A quick gist of the document defined in the mappings:
```
- propertyA <-- this property fill cause the issue, "myField" in the example
    - propertyA01
        - propertyA01_01
        - propertyA01_02
            - propertyA01_02_01
            - .. ~10 properties at this level
        - propertyA01_03
        - .. ~15 properties at this level
        - propertyA01_xx
    - propertyA02
    - .. ~85 properties at this level (with additional sub-levels)
    - propertyA80
- propertyB
- .. ~80 properties at this level
- propertyXX
```
 1. Create an index with no analysis, 
 2. Create a mapping with a document configured as above
 3. Query for the `_exists_` with the property pointed above

### PR DESCRIPTION
**Pull request by jrodewig**
Notes that you should increase the `indices.query.bool.max_clause_count`setting if you increase the `index.mapping.nested_objects.limit` setting.

Closes #46928.

### CODE REVIEW

File: docs/reference/mapping.asciidoc [hunk 1]
```diff
@@ -93,6 +93,13 @@ causing a mapping explosion:
 `index.mapping.nested_objects.limit`::
     The maximum number of `nested` JSON objects within a single document across
     all nested types, defaults to 10000.
++
+[IMPORTANT]
+====
+If you increase this setting, we recommend you also increase the
+<<search-settings,`indices.query.bool.max_clause_count`>> setting, which
+limits the maximum number of <<query-dsl-bool-query,boolean clauses>> in a query.
+====
```
Comment (jimczi) [inline docs/reference/mapping.asciidoc]: This limit is for the number of nested docs that are allowed on a single nested field. So by default we limit the number of nested documents in the array to 10,000 but that is not related to the boolean max clause. 
I'd put the same sentence but in the description of `index.mapping.total_fields.limit` with a warning that too many fields in mapping and queries should be avoided.

**Review by jimczi — COMMENTED** [review]
Thanks @jrodewig , I left one comment

File: docs/reference/mapping.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/mapping.asciidoc]: Thanks @jimczi. I've relocated the note to `index.mapping.total_fields.limit` with a2eff0807be.

**Review by jimczi — APPROVED** [review]
LGTM, thanks @jrodewig

**Comment by jrodewig** [comment]
@elasticmachine run elasticsearch-ci/oss-distro-docs
